src/Workspaces/Core/Portable/Workspace/Solution/TextLoader.cs

```diff
@@ -31,9 +35,19 @@ public abstract class TextLoader
         /// <exception cref="InvalidDataException"/>
         /// <exception cref="OperationCanceledException"/>
         public virtual Task<TextAndVersion> LoadTextAndVersionAsync(CancellationToken cancellationToken)
+        {
 #pragma warning disable CS0618 // Type or member is obsolete
-            => LoadTextAndVersionAsync(workspace: null, documentId: null, cancellationToken);
-#pragma warning restore CS0618
+            if (ImmutableInterlocked.GetOrAdd(
+                ref s_isObsoleteLoadTextAndVersionAsyncOverriden,
+                GetType(),
+                new Func<Workspace, DocumentId, CancellationToken, Task<TextAndVersion>>(LoadTextAndVersionAsync).Method.DeclaringType != typeof(TextLoader)))
```

Comment:
This should be using the overload that takes a `Func`, not using the direct result, as this will calculate the result every time no matter what.

Reply:
Of course :)